cluster/service/cluster_service.go

```diff
@@ -328,3 +328,20 @@ func (c clusterService) createIdentityCluster(ctx context.Context, identityID, c
 		return nil
 	})
 }
+
+// RemoveIdentityToClusterLink removes Identity to Cluster link/relation
+func (c clusterService) RemoveIdentityToClusterLink(ctx context.Context, identityID uuid.UUID, clusterURL string) error {
+	rc, err := c.Repositories().Clusters().LoadClusterByURL(ctx, clusterURL)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"cluster_url": clusterURL,
+			"err":         err,
+		}, "failed to load cluster with url %s", clusterURL)
+		return errors.NewBadParameterError("cluster-url", fmt.Sprintf("cluster with requested url %s doesn't exist", clusterURL))
```

Comment:
actually, this depends on the error you get: if it a `NotFoundError`, you can return the `BadParameterError`, otherwise, that should be an `InternalServerError`, I believe. See https://github.com/fabric8-services/fabric8-cluster/blob/ce2442e9c419398de7e2db3706de0b804702fbed/cluster/repository/cluster_repository.go#L111-L118

Reply:
This is excellent catch. Addressed in 83f9362